src/external_app/external_auth.ts

```diff
@@ -64,15 +64,21 @@ export class ExternalAuth extends Auth {
     });
   }
 
+  private _tokenCallbackPromise?: Promise<RefreshTokenResponse>;
+
   public async refreshAccessToken(force?: boolean) {
+    if (this._tokenCallbackPromise) {
```

Comment:
```suggestion
    if (this._tokenCallbackPromise && !force) {
```

We should create a new if  `force` is passed right?